msdk-featdet/msdk-featdet-ADAP-3D/src/main/java/io/github/msdk/featdet/ADAP3D/common/algorithms/PeakSimilarityTest.java

```diff
@@ -0,0 +1,231 @@
+/*
+ * (C) Copyright 2015-2017 by MSDK Development Team
+ *
+ * This software is dual-licensed under either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1 as published by the Free
+ * Software Foundation
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by the Eclipse Foundation.
+ */
+package io.github.msdk.featdet.ADAP3D.common.algorithms;
+
+import org.apache.commons.collections4.map.MultiKeyMap;
+
+import java.lang.Math;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * <p>
+ * PeakSimilarityTest Class.
+ * </p>
+ */
+public class PeakSimilarityTest {
+	
+	/**
+	 * <p>
+	 * PeakSimilarityResult class is used for returning lower and upper mz bound,boolean good peak value and list of similarity value.
+	 * </p>
+	 */
+	public static class PeakSimilarityResult{
+		List<Double> similarityValues;
+		boolean goodPeak;
+		int lowerMzBound;
+		int upperMzBound;
+	}
+
+	/**
+	 * <p>
+	 * peakSimilarityFunction method is used for getting similarity values for adjacent peaks, upper and 
+	 * lower mz bounds and decides whether the peak is good or bad.
+	 * @param objsliceSparseMatrix a {@link io.github.msdk.featdet.ADAP3D.common.algorithms.SliceSparseMatrix} object.
+	 * @param mz a {@link java.lang.Double} object.
+	 * @param leftBound a {@link java.lang.Integer} object.
+	 * @param rightBound a {@link java.lang.Integer} object.
+	 * @param fwhm a {@link java.lang.Double} object.
+	 * 
+	 * @return a {@link io.github.msdk.featdet.ADAP3D.common.algorithms.PeakSimilarityTest.PeakSimilarityResult} object
+	 * </p>
+	 */
+	public PeakSimilarityResult peakSimilarityFunction(SliceSparseMatrix objsliceSparseMatrix,double mz,int leftBound,int rightBound,double fwhm){
+		
+		double roundedFWHM = fwhm*10000;
+		int arrayCount = rightBound-leftBound+1;
+		
+		double[] normzlizedEICArray = new double[arrayCount];
```

Comment:
Hello Dr. Pluskal,
Thanks for suggestion. Yes. I was actually planning to add the explanation for each method as there are lot of calculation. I'll implement the comments.